ui/apps/everest/src/pages/databases/DbClusterView.utils.ts

```diff
@@ -97,3 +98,33 @@ export const getLastBackupTimeDiff = (lastBackup: Date): string => {
   return `${seconds} ${Messages.lastBackup.seconds} ${Messages.lastBackup.ago}
     `;
 };
+
+export const getLastBackupStatus = (
+  backups: Backup[],
+  schedules: Schedule[]
+) => {
+  if (!backups.length) {
+    if (!schedules.length) {
+      return Messages.lastBackup.inactive;
+    }
+    return Messages.lastBackup.scheduled;
+  }
+
+  const filteredBackups = backups.filter(
+    (backup) => backup.state !== BackupStatus.FAILED
+  );
+
+  const lastBackup = filteredBackups[filteredBackups.length - 1];
+
+  if (!lastBackup) {
+    return Messages.lastBackup.inactive;
+  }
+
+  if (lastBackup.state === BackupStatus.IN_PROGRESS) {
+    return Messages.lastBackup.pending;
+  }
+
+  if (lastBackup.state === BackupStatus.UNKNOWN) {
+    return Messages.lastBackup.notStarted;
+  }
```

Comment:
what if `lastBackup.state` is neither IN_PROGRESS or UNKNOWN? Won't this return anything?

Reply:
This function only handles the case in which we have no completed and successful backup to display in the Last Backup column.
`FAILED` status is filtered already, so if all of the backups are failed, we return 'Inactive'
And if there is any backup with `OK` status, we are in the first case, of displaying the time difference since the last completed backup ([here](https://github.com/percona/everest/pull/279/files#diff-8c361b4461716a80d07ed2591437fe86737824741ace373d991caf576f54d851L25-R32)), so all cases should be covered